### PR TITLE
make iconfonts work with latest qxcompiler in --target=build mode

### PR DESCRIPTION
**Pull request by oetiker**
It seems that something in the qxcompiler dependencies (babel?) has changed so that the large
`if else` structure in `Image.js` gets damaged in the compilation output when running in build mode. Switching to a `switch case` structure I was able to make icon fonts work again. 

As the code is much nicer like this anyway, I am submitting a patch. But as the `if else` structures do show up elsewhere in the code, further investigation into why this breaks would certainly be warranted. https://github.com/qooxdoo/qooxdoo-compiler/issues/207

### CODE REVIEW

File: framework/source/class/qx/ui/basic/Image.js [hunk 1]
```diff
@@ -396,26 +396,23 @@ qx.Class.define("qx.ui.basic.Image",
...
+          scale = true;
+          tagName = "div";
+          break;
+        case "nonScaled":
+          scale = false;
+          tagName = "div";
+          break;
+        default:
+          scale = true;
+          tagName = "img";
```
Comment (derrell) [inline framework/source/class/qx/ui/basic/Image.js]: Even though not strictly required here, the default block should have a trailing `break;` statement.

**Review by derrell — CHANGES_REQUESTED** [review]

**Review by cboulanger — DISMISSED** [review]

**Review by derrell — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by cajus — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Review by johnspackman — CHANGES_REQUESTED** [review]
On second thoughts, this should not be approved just for the sake of a bug in qxcompiler (especially as I think that may be fixed now).  Are there reasons *other than* https://github.com/qooxdoo/qooxdoo-compiler/issues/207 ?

**Comment by derrell** [comment]
Although certainly subjective, the `switch` statement feels to me like a cleaner implementation than a long series of `if/else` statements...

**Comment by johnspackman** [comment]
> Although certainly subjective, the `switch` statement feels to me like a cleaner implementation than a long series of `if/else` statements...

Fair enough, i'll remove my request for change

**Review by johnspackman — APPROVED** [review]

**Pull request merged**